lms/djangoapps/course_api/api.py

```diff
@@ -79,6 +81,9 @@ def course_detail(request, username, course_key):
         course_key,
     )
     overview.effective_user = user
+    if settings.FEATURES.get("ENABLE_V2_CERT_DISPLAY_SETTINGS") and auto_certificate_generation_enabled():
+        if not overview.certificates_display_behavior == CertificatesDisplayBehaviors.END_WITH_DATE:
+            overview.certificate_available_date = None
```

Comment:
I think this might be the last comment before I could approve the PR. I took the liberty of looking at exactly how this has been getting enabled in Studio and [this is how](https://github.com/openedx/edx-platform/blob/master/cms/templates/settings.html#L255-L299) is being enabled in Studio course dates page.

So what it does is, it checks if the waffle switch for certificates auto-generation is enabled it checks if a course is self-paced or instructor paced, If the course is instructor paced it enables the behavior right away that looks something like:
<img width="899" alt="image" src="https://user-images.githubusercontent.com/34372316/220099568-33c5377e-6afc-4a01-bf32-5f2026ea9a2b.png">

So it only checks the v2 settings as mentioned in the ticket https://github.com/mitodl/mitxonline/issues/1339 only to display multiple date options as seen in 
<img width="886" alt="image" src="https://user-images.githubusercontent.com/34372316/210965487-e4b63049-954f-457d-96ab-23a15f2dcb3b.png">
.

So I think we can make one last iteration and make it even simpler but simply calling https://github.com/openedx/edx-platform/blob/master/lms/djangoapps/certificates/api.py#L840 and adding the certificate available date to the API else None.

@aliraza-abbasi Let me know if it makes sense to you.

Reply:
Thank you @arslanashraf7. I think this would be much better. It makes sense to me as well. I'll update the code